### ISSUE TITLE
User Settable Number of Cores

### ISSUE THREAD

**Issue by jsnguyen** [ehpor/hcipy#229]
As far as I can tell, there isn't a way for the user to set the number of cores used in the FFT.

I've tried using `os.sched_setaffinity` but this isn't a great way of doing this because you can only set specific cores. This overrides the CPU scheduler and ends up being very slow and doesn't scale very well at all unless you use all the cores on the system.

My particular use case is that I'm running HCIPy on a shared use server with a bunch of cores, and I need to be able to limit the number of threads used by my processes.

Probably need to change the [file here.](https://github.com/ehpor/hcipy/blob/3cdb57f4d5c314dd1589fcd1cc6a669c82b9d39f/hcipy/_math/cpu.py#L3)

**Comment by ehpor** [comment]
You're right. With #213 we added backend support to be able to change the number of threads, but this implementation detail is not available from the `FastFourierTransform` class.

Indeed, the `get_num_available_cores()` function should be made a little smarter. Could you help me with finding out how, from Python, you can figure out the number of assigned cores to your job? The backend of Numpy already does this with environment variables, depending on the underlying implementation of BLAS. Does your cluster use any of these?

```
OMP_NUM_THREADS: openmp,
OPENBLAS_NUM_THREADS: openblas,
MKL_NUM_THREADS: mkl,
VECLIB_MAXIMUM_THREADS: accelerate,
NUMEXPR_NUM_THREADS: numexpr
```

Just searching for the nicest way of doing this.

**Comment by jsnguyen** [comment]
My cluster doesn't use any of those. The way I would solve this is that I would write a function that sets an environment variable like:

``` python
import os

def set_n_cpu(n_cpu: int):
    os.environ["HCIPY_N_CPU"] = f"{n_cpu}"

```

And check if it's set in `_math/cpu.py`. If not set then default to using all cores.
